main_plot.py

```diff
@@ -86,12 +158,38 @@ def filter_dataframe_by_dataset_method_and_subset(
     dataframe_of_results = report_loader()
     if dataframe_of_results is None:
         return pd.DataFrame()
-    dataframe_of_results = dataframe_of_results.loc[
-        dataframe_of_results["Dataset"] == dataset
-    ]
+
+    if dataset == "India city pollution":
+        dataframe_of_results = dataframe_of_results.loc[
+            dataframe_of_results["Dataset"] == "Indian city pollution"
+        ]
+    elif dataset == "stock_prices":
+        dataframe_of_results = dataframe_of_results.loc[
+            dataframe_of_results["Dataset"] == "Stock price"
+        ]
+    elif dataset == "airline_passengers":
+        dataframe_of_results = dataframe_of_results.loc[
+            dataframe_of_results["Dataset"] == "Airline passengers"
+        ]
+    elif dataset == "sun_spots":
+        dataframe_of_results = dataframe_of_results.loc[
+            dataframe_of_results["Dataset"] == "Sun spots"
+        ]
+
+    elif dataset == "list_of_tuples":
+        dataframe_of_results = dataframe_of_results.loc[
+            dataframe_of_results["Dataset"] == "Straight line"
+        ]
+
+    else:
+        dataframe_of_results = dataframe_of_results.loc[
+            dataframe_of_results["Dataset"] == dataset
+        ]
+
```

Comment:
This might be improved by using a dictionary (I asked ChatGPT to suggest this, you can have a go with ChatGPT yourself too!):
`dataset_to_string = {
    "India city pollution": "Indian city pollution",
    "stock_prices": "Stock price",
    "airline_passengers": "Airline passengers",
    "sun_spots": "Sun spots",
    "list_of_tuples": "Straight line"
}

if dataset in dataset_to_string:
    dataframe_of_results = dataframe_of_results.loc[
        dataframe_of_results["Dataset"] == dataset_to_string[dataset]
    ]
else:
    dataframe_of_results = dataframe_of_results.loc[
        dataframe_of_results["Dataset"] == dataset
    ]
`